proto/provenance/metadata/v1/query.proto

```diff
@@ -88,11 +89,42 @@ service Query {
   rpc RecordSpecificationByID(RecordSpecificationByIDRequest) returns (RecordSpecificationByIDResponse) {
     option (google.api.http).get = "/provenance/metadata/v1/recordspec/id/{record_specification_id}";
   }
+
+  // ---- OS locator methods -----
+  // Params queries all parameters for os locator sub module.
+  rpc OSLocatorParams(OSLocatorParamsRequest) returns (OSLocatorParamsResponse) {
+    option (google.api.http).get = "/provenance/metadata/v1/locator/params";
+  }
+
+  // for querying oslocator by address.
+  rpc OSLocator(OSLocatorRequest) returns (OSLocatorResponse) {
+    option (google.api.http).get = "/provenance/metadata/v1/locator/{owner}";
+  }
+
+  // for querying oslocator by uri.
+  rpc OSLocatorByURI(OSLocatorByURIRequest) returns (OSLocatorByURIResponse) {
+    option (google.api.http).get = "/provenance/metadata/v1/locator/uri/{uri}";
```

Comment:
this seems like a bad idea since you have a uri in a uri here ... do we need this query endpoint at all? might be best to just remove it and save future headaches.

Reply:
{uri} is just the path param,  i can remove the endpoint, was thinking a client might want to query all the addresses it has pointing to it's object store. 🤷